addons/html_builder/static/src/website_mail_group/mail_group_option_plugin.js

```diff
@@ -0,0 +1,95 @@
+import { InputConfirmationDialog } from "@html_builder/snippets/input_confirmation_dialog";
+import { Plugin } from "@html_editor/plugin";
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+
+class MailGroupOptionPlugin extends Plugin {
+    static id = "mailGroupOption";
+    static dependencies = ["builderActions"];
+    resources = {
+        builder_options: [
+            {
+                template: "website_mail_group.MailGroupOption",
+                selector: ".s_group",
+            },
+        ],
+        dropzone_selector: {
+            selector: ".s_group",
+            dropNear: "p, h1, h2, h3, blockquote, .card",
+        },
+        builder_actions: {
+            mailGroupAction: {
+                apply: ({ editingElement, value }) => {
+                    const { id } = JSON.parse(value);
+
+                    this.dependencies.builderActions
+                        .getAction("dataAttributeAction")
+                        .apply({ editingElement, param: { mainParam: "id" }, value: id });
+                },
+                clean: ({ editingElement }) => {
+                    this.dependencies.builderActions
+                        .getAction("dataAttributeAction")
+                        .clean({ editingElement, param: { mainParam: "id" } });
+                },
+                getValue: ({ editingElement }) => {
+                    const value = {};
+                    const id = this.dependencies.builderActions
+                        .getAction("dataAttributeAction")
+                        .getValue({ editingElement, param: { mainParam: "id" } });
+                    if (!id) {
+                        return;
+                    }
+                    value.id = parseInt(id);
+                    return JSON.stringify(value);
+                },
+            },
+            createMailGroup: {
+                load: ({ editingElement, value }) => this.createGroup(value),
+                apply: ({ editingElement, loadResult: id }) => {
+                    if (id) {
+                        this.dependencies.builderActions
+                            .getAction("mailGroupAction")
+                            .apply({ editingElement, value: JSON.stringify({ id }) });
+                    }
+                },
+            },
+        },
+        on_snippet_dropped_handlers: this.onSnippetDropped.bind(this),
+    };
+
+    async onSnippetDropped({ snippetEl }) {
+        if (snippetEl.dataset.snippet !== "s_group") {
+            return;
+        }
+        const group = await this.services.orm.call("mail.group", "name_search", [""], { limit: 1 });
+        const id = group[0]?.[0] || (await this.createGroup());
```

Comment:
the editor could have been destroyed here. so maybe useful to protect agains this:
```js
if (this.isDestroyed) {
  return;
}
```